packages/models/src/resource-binding/cell-change-buffer.ts

```diff
@@ -0,0 +1,56 @@
+import { Input } from '@ckb-lumos/base'
+import { ActorURI } from '../actor'
+import { ResourceBindingRegistry } from './interface'
+import { CellChangeData } from './types'
+
+export class CellChangeBuffer {
+  #buffer: Map<ActorURI, [ResourceBindingRegistry, Input[], CellChangeData[]][]> = new Map()
+  #readyList: Array<ActorURI> = []
+
+  hasReadyStore(): boolean {
+    return this.#readyList.length > 0
+  }
+
+  push(uri: ActorURI, data: [ResourceBindingRegistry, Input[], CellChangeData[]]) {
+    let buffer = this.#buffer.get(uri)
+    if (!buffer) {
+      buffer = []
+    }
+    buffer.push(data)
+    this.#buffer.set(uri, buffer)
+  }
+
+  pop(): [ResourceBindingRegistry, Input[], CellChangeData[]][] | undefined {
+    const uri = this.#readyList.pop()
+    if (uri) {
+      const changes = this.#buffer.get(uri)
+      this.#buffer.delete(uri)
+      return changes
+    }
+  }
+
+  popAll(): [ResourceBindingRegistry, Input[], CellChangeData[]][][] {
+    let length = this.#readyList.length
+    const res: [ResourceBindingRegistry, Input[], CellChangeData[]][][] = []
+    while (length-- > 0) {
+      const data = this.pop()
+      if (data) {
+        res.push(data)
+      }
+    }
```

Comment:
Use `this.#readyList.forEach` instead of a `while loop`

Ref: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach

Reply:
> > Use `this.#readyList.forEach` instead of a `while loop`
> > Ref: [developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach)
> 
> `Store`s are popped from ready list one by one, so I use `while loop` instead of `forEach`. Another reason is I want to process a certain number of `Store` in ready list each round.

I didn't get the point because `forEach` handles elements one by one, too. And `forEach` could iterate a certain number by **array slice**.